public/main.js

```diff
@@ -65,7 +65,20 @@ function searchDict(query) {
       return true
     }
   })
-  results.sort((a, b) => b.percentile - a.percentile)
+  results.sort((a, b) => {
+    // prioritizes exact matches
+    const aExact = a.searchablePinyin === query, bExact = b.searchablePinyin === query;
+    if (aExact && !bExact) return -1;
+    if (!aExact && bExact) return 1;
```

Comment:
Could you also add exact matching support for definitions? That one might be trickier because the definitions are newline-separated. So you could match on each item from `a.definition.split('\n')` - though that might be slow enough to warrant a more elegant search method.

Reply:
So I added simple logic to see if the definitions include the string phrase by splitting the definitions, and it is a little slow for inputs of 2 or fewer characters. Inputs of 3 or higher seem to have the results filtered enough that it can sort them quickly enough to feel responsive.

I'll think about alternatives or other solutions. Maybe adding some kind of cache would help so that it doesn't have to re parse the dictionaries every time it's sorting.

Reply:
I made the requested changes and used a cache to speed things up a little bit. Chrome seems responsive for all length entries. Firefox has a tiny hang with 1 letter, but has sped up on 2 letter or more.